src/recon_buildblock/Parallelproj_projector/ParallelprojHelper.cxx

```diff
@@ -73,7 +73,7 @@ detail::ParallelprojHelper::ParallelprojHelper(const ProjDataInfo& p_info, const
   Bin bin;
   LORInAxialAndNoArcCorrSinogramCoordinates<float> lor;
   LORAs2Points<float> lor_points;
-  const float radius = p_info.get_scanner_sptr()->get_inner_ring_radius();
+  const float radius = p_info.get_scanner_sptr()->get_outer_FOV_radius() + p_info.get_scanner_sptr()->get_average_depth_of_interaction();
```

Comment:
Good point, that should not actually be needed. Will test it without and change.

Reply:
The DOI is actually required - without we get the horizontal streaks again in the boundary. This is because internally the effective_ring_radius is used when we call get_LOR on the ProjDataInfo class.

Reply:
This is conceptually a bit weird. On the other hand, I notice that the `VoxelsOnCartesianGrid` constructor uses essentially `max(abs(get_s()))` for the default image radius:
https://github.com/UCL/STIR/blob/da4e6e6cd3c9acf2091a89a42283aeba8cc43380/src/buildblock/VoxelsOnCartesianGrid.cxx#L290-L298
That strategy initially makes some sense. If the scanner has a larger FOV, but the data is truncated, it tries to reconstruct a smaller image. However, it isn't necessarily correct (the object might still be larger), hence using `get_max_FOV_radius()` there would be more correct. That would be a major compatibility breaking change though.

For #1070 however, the question is if `get_max_FOV_radius()` is larger than the image radius, or probably more importantly `max(abs(get_s()))`. I guess in the latter case, we might still have the problem that the LOR becomes undefined, and we don't see/catch it?

Reply:
True, using get_max_FOV_radius() would be more correct there.

We only see this problem with the undefined LOR and cylinder intersections when get_max_FOV_radius + DOI is smaller than the image radius, and I think this should never happen (we would never attempt to reconstruct stuff behind the detectors). get_max_FOV_radius can be much larger, and parallelproj would still give us correct results.

Reply:
I suspect it isn't the image radius though. Could you check what parallelproj does if you give it a small radius?

Instead I think we should check the return value of `get_intersections_with_cylinder` such that we don't ever use intersections with a cylinder where there aren't any,